php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -832,15 +834,27 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         }
         $newData = array_merge($oldData ?? [], $values);
 
-        // Save the JSON to the flag.Data column.
-        //
-        // json_encode ensures that this is safe. If we use the safe wrapper,
-        // HTML encoding the quotation marks will make it invalid JSON.
-        $db->unsafeUpdate(
+        // If there's already a row with the same data, re-use the DataID.
+        // Otherwise, insert it, then update the DataID on flag.
+        $jsonencoded = json_encode($newData);
+        $newDataID   = $db->pselectOne(
```

Comment:
Note: I had to decide between creating a new DataID or updating the Data for that DataID. I decided to go with inserting a new DataID so that entries with the same content (ie. DDE instrument, or candidates which happen to have the same answers on a short questionaire) can be de-duplicated.